### ISSUE TITLE
Format Field not working on import

### ISSUE THREAD

**Issue by crisr15** [notch8/atla_digital_library#231]
# Summary

TLDR: After bulkrax upgrade, on CSV imports, the format (digital) field was showing the original imported MIME types instead of the terms that were set up to be used by ATLA in their [formats.yml](https://github.com/scientist-softserv/atla_digital_library/blob/main/config/authorities/formats.yml) authority. 

it looks like something is not working in our Format filed - this audio/mpeg should have been converted. In our live site this collection was already imported with this field being correct. https://dl-staging.atla.com/concern/works/q237hs04k?locale=en and here it is on the live site https://dl.atla.com/concern/works/fq9786682?locale=en

example: audio/mpeg should be converted to say MPEG
here are all of the types that should show up
<img width="312" alt="image" src="https://user-images.githubusercontent.com/73361970/234613889-e1c37b27-6ce8-4b30-a471-a91139052876.png">

# Testing Instructions
1. Go to dashboard on staging : [dl-staging](https://dl-staging.atla.com/)
2. Update/re-run the following importer: https://dl-staging.atla.com/importers/64?locale=en
3. make sure you reharvest all items, i.e. remove them all and re-create them. this is the bottom button option when updating the importer
4. After the works are done importing, check on the work show page that the format (digital) of the items in the importer say "MPEG" and not "audio/mpeg"

# Acceptance Criteria
- [ ] As a user, when i upload a CSV, IANA (MIME) type name is being converted correctly in the Format (Digital) field

# Notes
This documentation has the rules for Field adjustments on import and order: https://docs.google.com/spreadsheets/d/1naH5DSZc6-RWCGjYxU2zNsYRxN8oAKPMbag-s16tR3c/edit#gid=35320470

We need to look at the previous parsing on the Format (Digital) field for the CSV parser and make sure that the IANA (MIME) type name is being used.

Importer on staging where issue can be seen: https://dl-staging.atla.com/importers/64?locale=en

**Comment by labradford** [comment]
Is there a way to tell which importer was used to add this item?

**Comment by crisr15** [comment]
Slack conversation for clarification on this ticket: https://assaydepot.slack.com/archives/C031E2NF43T/p1681855582693109

**Comment by ckarpinski** [comment]
I think on staging it is this importer https://dl-staging.atla.com/importers/64?locale=en

**Comment by crisr15** [comment]
This documentation has the rules for Field adjustments on import and order:  https://docs.google.com/spreadsheets/d/1naH5DSZc6-RWCGjYxU2zNsYRxN8oAKPMbag-s16tR3c/edit#gid=35320470

We need to look at the previous parsing on the Format (Digital) field for the CSV parser and make sure that the IANA (MIME) type name is being used.

**Comment by summer-cook** [comment]
# Notes
- The CSV parser is the exact same for bulkrax between the version that is on atla prod (`atla_tmp` branch of bulkrax) and the current version of bulkrax. Both of them have an empty cvs parser, so the defaults are generated automatically. 
- I would be curious to see if the formats are generated correctly for this CSV in prod, because if they are not, i do not think this is a bulkrax related issue.

**Comment by summer-cook** [comment]
# More Notes

- I tried to upload the CSV from the aforementioned importer as a new/private test importer on prod to see if the format was parsing correctly. All items on this csv failed to import on prod with a name error: 
- I tried with the exact same csv on staging and it DID import correctly.
- On both prod & main, the only place this format parsing is happening is in [this format method](https://github.com/scientist-softserv/atla_digital_library/blob/5cf6724c3688caa25187f669fa44dad332cb391d/app/matchers/bulkrax/atla_matcher.rb#L7), which calls a service from hyrax. its possible the service needs to be updated. I think this has been broken for a long time, because from what I can tell, the method is not actually being called anywhere.

**Comment by summer-cook** [comment]
@ckarpinski so just to confirm, formats that say `MPEG` should be converting to say `audio/mpeg` - not the other way around

In the examples provided, format already imports to be called `audio/mpeg` on staging. On a [test import with the csv from the importer that you provided](https://dl-staging.atla.com/importers/74?locale=en), i see all the formats with `audio/mpeg`,  the work on prod just says MPEG.
this [list of IANA mime types](https://www.iana.org/assignments/media-types/media-types.xhtml) from iana.org shows that the type name is audio/mpeg. I feel like i may be missing something?

**Comment by ckarpinski** [comment]
No its the other way around. You can see what we use for formats by going here https://dl.atla.com/catalog?utf8=%E2%9C%93&q= and looking at the facet for format digital.

**Comment by ckarpinski** [comment]
What should be happening is if the metadata says mpeg, MPEG or Mpeg or whatever OR if it says audio/mpeg - all these should be output to MPEG

**Comment by summer-cook** [comment]
Ok- that's what i originally thought but I got confused when i looked at the MIME types documentation! thank you

**Comment by DiemBTran** [comment]
**This passes SoftServ QA:** See attached screenshots.

tested on:
- importer: https://dl-staging.atla.com/importers/64?locale=en
- work from importer: https://dl-staging.atla.com/concern/works/08612n733?locale=en

<details><summary>"Update and Replace Files" updated the works' format (digital) from "audio/mpeg" to "MPEG"</summary>

![Atla Digital Library 2023-04-27 at 2 33 42 PM](https://user-images.githubusercontent.com/29311858/234997996-91831b4d-d6d1-41fb-af62-23315ec4fb4c.jpg)
![Blessing of the Bicycles servicesermon - Atla Digital Library 2023-04-27 at 2 38 34 PM](https://user-images.githubusercontent.com/29311858/234998009-c671b19c-1904-4d0d-a3c5-f6747f26c7c0.jpg)
![Blessing of the Bicycles service - sermon - Atla Digital Library 2023-04-27 at 2 39 34 PM](https://user-images.githubusercontent.com/29311858/234998023-4985562e-8019-4a53-8e37-219cc63b4594.jpg)
</details>

**Comment by alishaevn** [comment]
@ckarpinski this is ready for your review!

**Comment by ckarpinski** [comment]
Here is the original list shared with you all for this field - it includes more than what is listed in the opening screenshot https://docs.google.com/document/d/1Ubn7DHb5Ay33zlHR-tE07T8WUF-q4s4PJf2xtswTrWc/edit

@summer-cook  Can you let me know if all of these are set up to work?

**Comment by ckarpinski** [comment]
Right now on staging there are JPG and jpg and jpgeg and MP$ and mp4 - is this just left over mess or is that how this is set up? all variations of JPG etc should be the same on output

**Comment by summer-cook** [comment]
@ckarpinski 
1. Yes, they are all set up to work. [This is the file](https://github.com/scientist-softserv/atla_digital_library/blob/main/config/authorities/formats.yml) the formats pull from that the app has been using for the past 4 years. That file had been disconnected with the bulkrax upgrade, and now it is connected to the parser. 
2. those are leftover. This will not fix previously imported works, it will only apply to newly imported works, since the data will need to be parsed as it comes in.

**Comment by ckarpinski** [comment]
Thank you for the explanation of how it stopped working.
